geniza/corpus/management/commands/add_links.py

```diff
@@ -255,32 +272,82 @@ def add_link(self, row):
         doc = Document.get_by_any_pgpid(int(row["object_id"]))
         if not doc:
             self.stats["document_not_found"] += 1
+            self.not_found_documents += [row]
             if self.verbosity > self.v_normal:
                 self.stdout.write(
                     "Document %s not found in database" % row["object_id"]
                 )
             return
 
         # target url: combine base url for this link type with link target from csv
-        target_url = "".join([self.base_url[link_type], row["link_target"]])
+        target_url = "".join([self.base_url[link_type], quote(row["link_target"])])
         # get doc relation for this link type
         doc_relation = self.document_relation[link_type]
 
-        # get source based on the link type
+        # get source and location based on the link type
         if link_type == "goitein_note":
             source = self.get_goitein_source(doc=doc, title="typed texts")
+            base_name = basename(row["link_target"])
+            try:  # use base name up to PGPID for goitein texts
+                end_index = base_name.index("(PGPID")
+            except ValueError:  # if PGPID not present, use base name up to extension
+                end_index = base_name.rindex(".") if "." in base_name else 0
+            # use string up to end_index and strip whitespace for location
+            location = base_name[:end_index].strip() if end_index else base_name.strip()
```

Comment:
let's simplify and just use `split`, which will work whether the text is there or not
```suggestion
            # use target location up to parenthetical pgpid
            location = basename.split("(PGPID")[0].strip()
```